frontend/src/container/MetricsApplication/MetricsPageQueries/OverviewQueries.ts

```diff
@@ -22,79 +24,62 @@ import {
 export const latency = ({
 	servicename,
 	tagFilterItems,
+	spanMetricFlg = false,
 }: LatencyProps): QueryBuilderData => {
-	const autocompleteData: BaseAutocompleteData[] = [
-		{
-			key: WidgetKeys.DurationNano,
-			dataType: DataType.FLOAT64,
-			isColumn: true,
-			type: MetricsType.Tag,
-		},
-		{
-			key: WidgetKeys.DurationNano,
-			dataType: DataType.FLOAT64,
-			isColumn: true,
-			type: MetricsType.Tag,
-		},
+	const newAutoCompleteData: BaseAutocompleteData = {
+		key: spanMetricFlg
+			? WidgetKeys.Signoz_latency_bucket
+			: WidgetKeys.DurationNano,
+		dataType: DataType.FLOAT64,
+		isColumn: true,
+		type: spanMetricFlg ? null : MetricsType.Tag,
+	};
+
+	const autocompleteData: BaseAutocompleteData[] = Array(3).fill(
+		newAutoCompleteData,
+	);
+
+	const filterItem: TagFilterItem[] = [
 		{
-			key: WidgetKeys.DurationNano,
-			dataType: DataType.FLOAT64,
-			isColumn: true,
-			type: MetricsType.Tag,
+			id: '',
+			key: {
+				key: spanMetricFlg ? WidgetKeys.Service_name : WidgetKeys.ServiceName,
+				dataType: DataType.STRING,
+				type: spanMetricFlg ? MetricsType.Resource : MetricsType.Tag,
+				isColumn: !spanMetricFlg,
+			},
+			op: spanMetricFlg ? OPERATORS.IN : OPERATORS['='],
+			value: spanMetricFlg ? [servicename] : servicename,
```

Comment:
This can be `IN` for both cases, with the value being array `[servicename]` ie. `service_name IN ['redis']` is the same as `service_name = 'redis'` when there is only one value.

Reply:
Keeping is as it is as discussed offline.